mwclient/errors.py

```diff
@@ -14,6 +14,10 @@ class MaximumRetriesExceeded(MwClientError):
     pass
 
 
+class MwClientNotFound(MwClientError):
```

Comment:
Hm, this name implies that it's more generic than it currently is. Do you see other use cases for it? If not, I think it's better to be specific and name it `UserNotFound`. Especially since the other exceptions are generally quite specific (and do not use the Mw prefix).

Reply:
I renamed `MwClientNotFound` to `UserNotFound` and creates new ones to get
  better debugging possibilities:
    * `NotFound`
    * `CreateError`
    * `UserCreateError`